src/contexts/Theme.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react'
+import { ThemeProvider } from '@mui/material/styles'
+import { dark, light } from '@/plugins/muiTheme'
+
+interface Children {
+  children: React.ReactNode
+}
+
+// interface ColorModeContextValue {
+//   mode: 'light' | 'dark'
+//   colorMode: {
+//     toggleColorMode: () => void
+//   }
+//   theme: Theme
+// }
+
+const ColorModeContext = React.createContext({ toggleColorMode: () => {} })
+
+export default function ToggleColorMode({ children }: Children) {
+  const [mode, setMode] = React.useState<'light' | 'dark'>('light')
+  const colorMode = React.useMemo(
+    () => ({
+      toggleColorMode: () => {
+        setMode((prevMode) => (prevMode === 'light' ? 'dark' : 'light'))
+      }
+    }),
+    []
+  )
+
+  const theme = React.useMemo(() => (mode === 'light' ? light : dark), [mode])
+
+  const value: any = {
+    // const value: ColorModeContextValue = {
+    mode,
+    colorMode,
+    theme
+  }
```

Comment:
we will use this to pass the correct values, functions to the provider
```ts
const value: ColorModeContextValue = {
  mode,
  toggleColorMode: colorMode.toggleColorMode
}
```

Reply:
Fixed